From a3c158dad214a60726cf1332021b083802ac59f9 Mon Sep 17 00:00:00 2001
From: Vasily Litvinov <45396231+vnlitvin@users.noreply.github.com>
Date: Fri, 22 Mar 2019 19:18:37 +0300
Subject: [PATCH] Safe version of ascii macros, add missing tolower_ascii
 (#25836)

* Safer version of ascii macros in portable.h

* Add missing tolower_ascii macro needed by lowercase() in parse_helper.h
---
 pandas/_libs/src/headers/portable.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/pandas/_libs/src/headers/portable.h b/pandas/_libs/src/headers/portable.h
index 9ac4ebc306ba..1976addace3f 100644
--- pandas/_libs/src/headers/portable.h
+++ pandas/_libs/src/headers/portable.h
@@ -7,8 +7,9 @@
 
 // GH-23516 - works around locale perf issues
 // from MUSL libc, MIT Licensed - see LICENSES
-#define isdigit_ascii(c) ((unsigned)c - '0' < 10)
-#define isspace_ascii(c) (c == ' ' || (unsigned)c-'\t' < 5)
-#define toupper_ascii(c) (((unsigned)c-'a' < 26) ? (c & 0x5f) : c)
+#define isdigit_ascii(c) (((unsigned)(c) - '0') < 10u)
+#define isspace_ascii(c) (((c) == ' ') || (((unsigned)(c) - '\t') < 5))
+#define toupper_ascii(c) ((((unsigned)(c) - 'a') < 26) ? ((c) & 0x5f) : (c))
+#define tolower_ascii(c) ((((unsigned)(c) - 'A') < 26) ? ((c) | 0x20) : (c))
 
 #endif
